warp-core/src/main/scala/com/workday/warp/persistence/CoreQueries.scala

```diff
@@ -562,4 +562,14 @@ trait CoreQueries extends AbstractQueries {
   override def updateTestExecutionThreshold[T: TestExecutionRowLikeType](testExecution: T, newThreshold: Double): DBIO[Int] = {
     TestExecution filter { _.idTestExecution === testExecution.idTestExecution } map { _.responseTimeRequirement } update newThreshold
   }
+
+
+  override def getPriorTestExecutionsQuery[T: TestExecutionRowLikeType](testExecution: T,
+                                                                        limit: Int): DBIO[Seq[TablesLike.TestExecutionRowLike]] = {
+    TestExecution
+      .filter(exec => exec.idTestDefinition === testExecution.idTestDefinition && exec.idTestExecution < testExecution.idTestExecution)
+      .sortBy(_.idTestExecution.desc)
+      .take(limit)
```

Comment:
Following up from my prior comment, intuitively for a collection this should take nothing (`List(1,2,3,4) take -2` should just be `List()`), but is that the same behavior we see here? Might just be better to change [this](https://github.com/Workday/warp-core/pull/98/files#diff-cea90ea65ca3da23111e3ca4adb902cdc2c7f85f5f51405e3e33525de68bab81R55) to avoid dropping down here at all

Reply:
i'll change that. slick converts this to a sql `LIMIT`, but we should avoid making a db call at all if its not necessary.